src/data/qpf/indexed/basic.lean

```diff
@@ -0,0 +1,458 @@
+/-
+Copyright (c) 2020 Simon Hudon. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Simon Hudon
+-/
+import data.pfunctor.indexed.basic
+import tactic.mk_opaque
+universe u
+
+/-!
+
+# Indexed quotients of polynomial functors.
+
+Basic definition of indexed QPF. QPFs form a compositional framework
+for defining inductive and coinductive types, their quotients and nesting.
+
+The idea is based on building ever larger functors. For instance, we can define
+a list using a shape functor:
+
+```lean
+inductive list_shape (a b : Type)
```

Comment:
Does anyone else care that this goes against the usual convention of using Greek letters for types? (I can live with it if nobody else minds.)

Reply:
The mathematician cadre has been working hard at killing that convention, but I think it would be better to use greek rather than lowercase letters in this instance. (I can't think of any example of a conventional type name that is a lowercase letter in lean/mathlib. Even `k` for fields is blackboard bolded first.) Here the choice just looks like the consequence of an impoverished keyboard.

Reply:
>  impoverished keyboard.

indeed.

> The mathematician cadre has been working hard at killing that convention

How have they been doing that?

Reply:
When a type is immediately being equipped with an algebraic typeclass (e.g. `ring`), we're using one of the letters that mathematicians would conventionally use to denote such a (bundled) object (e.g. `R`).

For mathematicians, `M` is a manifold or a module, `R` is a ring`, `C` is a category, and so on.

(Mathematicians like to make fun of physicists for doing this, but we do it too.)